src/ja/piccoma/src/eu/kanade/tachiyomi/extension/ja/piccoma/Piccoma.kt

```diff
@@ -0,0 +1,245 @@
+    }
+
+    override fun searchMangaParse(response: Response): MangasPage {
+        if (response.request.url.toString().contains("/ranking/")) {
+            return popularMangaParse(response)
+        }
+
+        val result = response.parseAs<SearchResponseDto>()
+        val mangas = result.data.products.map {
+            SManga.create().apply {
+                url = "/web/product/${it.id}"
+                title = it.title
+                thumbnail_url = "https:${it.img}"
+            }
+        }
+        val currentPage = response.request.url.queryParameter("page")!!.toInt()
+        val hasNextPage = currentPage < result.data.totalPage
+        return MangasPage(mangas, hasNextPage)
+    }
+
+    override fun mangaDetailsParse(response: Response): SManga {
+        val document = response.asJsoup()
+        return SManga.create().apply {
+            title = document.selectFirst("h1.PCM-productTitle")!!.text()
+            author = document.select("ul.PCM-productAuthor li a").joinToString { it.text() }
+            genre = document.select("ul.PCM-productGenre li a, .PCM-productDesc_tagList li a").joinToString { it.text() }
+            description = document.selectFirst("div.PCM-productDesc > p")?.text()
+            document.selectFirst("img.PCM-productThum_img")?.absUrl("src")?.let { thumbnail_url = it }
+            status = when {
+                document.selectFirst("ul.PCM-productStatus")?.text()
+                    ?.contains("連載中") == true -> SManga.ONGOING
+
+                document.selectFirst("ul.PCM-productStatus")?.text()
+                    ?.contains("完結") == true -> SManga.COMPLETED
+
+                else -> SManga.UNKNOWN
+            }
+        }
+    }
+
+    override fun chapterListRequest(manga: SManga): Request {
+        return GET("$baseUrl${manga.url}/episodes?etype=E", headers)
+    }
+
+    override fun chapterListParse(response: Response): List<SChapter> {
+        val document = response.asJsoup()
+        val episodes = document.selectFirst("ul#js_episodeList")!!
+
+        return episodes.select("li").map {
+            val link = it.selectFirst("a")!!
+            val productId = link.attr("data-product_id")
+            val episodeId = link.attr("data-episode_id")
+            val titleElement = it.selectFirst("div.PCM-epList_title h2")?.text()
+            val statusElement = it.selectFirst("div.PCM-epList_status")
+
+            val isPoint = statusElement?.selectFirst(".PCM-epList_status_point") != null
+            val isWaitFree = statusElement?.selectFirst(".PCM-epList_status_waitfree") != null
+            val isZeroPlus = statusElement?.selectFirst(".PCM-epList_status_zeroPlus") != null
+
+            val icon = when {
+                isPoint -> "🔒 "
+                isWaitFree || isZeroPlus -> "➡️ "
+                else -> ""
+            }
+
+            SChapter.create().apply {
+                url = "/web/viewer/$productId/$episodeId"
+                name = "$titleElement $icon"
```

Comment:
Wrong order? `icon` has a space after the emoji. Not sure since you're specifically looking for 
```suggestion
                name = "$icon$titleElement"
```